src/KubeOps/Operator/OperatorSettings.cs

```diff
@@ -78,5 +79,7 @@ public sealed class OperatorSettings
         /// <para>The search will be performed on each "Start" of the controller.</para>
         /// </summary>
         public bool PreloadCache { get; set; }
+
+        internal JsonSerializerSettings SerializerSettings { get; set; } = null!;
```

Comment:
instead of setting this to `null!` (which is just my personal opinion), would it suffice to add the JsonSerializerSettings as singleton to the dependency injection and then just inject the settings when needed?